### ISSUE TITLE
AnimationPlayer.is_playing() not showing correct status in _process

### ISSUE THREAD

**Issue by gitnob** [godotengine/godot#3716]
tested the following sequence:
```
print(is_playing())   ->  "false"
play("anyAnimation")   ->  "true"
// while animation is palying
print(is_playing())   ->  "true"
// after animation is stopped
print(is_playing())   ->  "true"      error!!
```

seems to not be triggered back to false status after animation is finished.
(Debian jessie amd_64 - godot 2.0 RC1)

**Comment by akien-mga** [comment]
I can't reproduce this, if I print ```is_playing()``` in ``_fixed_process`` after having started a non-looping animation, it does print "False" once the animation stops playing. I guess your animation is set to loop or to trigger another looping animation?

At any rate a minimal project that reproduces the issue would be useful to understand it better.

**Comment by volzhs** [comment]
I can't reproduce either for this. (tested with ad405ade17630c927e06530dbbaad4e730193774)
This is what I did.
```gdscript
func _ready():
	get_node("AnimationPlayer").connect("finished", self, "on_anim_finished")
	get_node("AnimationPlayer").play("New Anim")
	print("is_playing ? ", get_node("AnimationPlayer").is_playing())

func on_anim_finished():
	print("is_playing ? ", get_node("AnimationPlayer").is_playing())
```

this prints
```
is_playing ? True
is_playing ? False
```

**Comment by gitnob** [comment]
It should also work in _process() while in IDLE mode of the animation player. But it does not. Just try to put your code into the _process() or try to use mine.
In the animFish.gd script you can uncomment the necessary line to check the difference.
BTW: Within _fixed_process() my code is working fine.
IIRC t is not mentioned in the documentation to use the _fixed_process() method for this check nor does it make any sense that you have to use is_playing() only in that method, though IMO a bug.
But I am willing to learn... ;-)

[fish01.gd.zip](https://github.com/godotengine/godot/files/131369/fish01.gd.zip)

**Comment by gitnob** [comment]
That's right volzhs. Is working also fine for me within the _ready() and _on_finished() method, but not in the _process() method after a non-looped animation stopped.

**Comment by akien-mga** [comment]
Indeed it does not seem to work properly in ``_process``, which is probably a bug.

**Comment by gitnob** [comment]
BTW set_speed does not work neither ... another bug.

**Comment by akien-mga** [comment]
Why close this one though?

**Comment by gitnob** [comment]
Oh sorry...

**Comment by trollworkout** [comment]
also related .stop() is not actually working properly. it should reset animation to frame 1 but it doesn't . animation stays at current frame. Dunno if this was fixed yet.

**Comment by akien-mga** [comment]
That's unrelated @trollworkout, please open another issue for that.

**Comment by DualMatrix** [comment]
This seems to be fixed in master.
